src/Facility.Core/Http/HttpClientService.cs

```diff
@@ -101,6 +104,8 @@ protected HttpClientService(HttpClientServiceSettings? settings, Uri? defaultBas
 			{
 				var contentType = mapping.RequestBodyContentType ?? requestHeaders?.GetContentType();
 				httpRequest.Content = GetHttpContentSerializer(requestBody.GetType()).CreateHttpContent(requestBody, contentType);
+				if (m_enableRequestCompression)
```

Comment:
It would be nice to only do this if the content is longer than, say, 256 bytes (because it's probably not worth it for small payloads) but I don't know of a way to efficiently try to get the content length for arbitrary HttpContent.

Reply:
I think that in the general case, I don't want to read the request body (which could be expensive?) before it's being sent. And then, once it's being sent, I assume that the headers have already been processed and we can't optionally add `Content-Encoding`? 

Like I said, this would be a "nice to have", so I don't feel strongly about implementing it now.

Reply:
Yep, that's fine. To be clear, my suggestion would be to read the 256-byte header in `CompressingHttpContent.CreateAsync`. If the header is all there is, we don't set the content encoding. `SerializeToStreamAsync` respects that by writing the header directly when not compressing, or first writing the header to the compressing stream before copying the rest.